### PR TITLE
Add GitHub Release creation to release skill

### LINKED ISSUE #222: Release skill should create a GitHub Release after tagging
## Problem

The release skill (step 10) ends after creating an annotated git tag and reminding the user to push. It does not create a GitHub Release, which is a separate concept from a git tag. This means the GitHub Releases page stays out of date unless the user manually runs `gh release create`.

## Proposed solution

Add a step after tagging (or as part of the push reminder) that creates a GitHub Release using the CHANGELOG entries for the new version. This could be:

- An optional step 11 that runs `gh release create` with the version's changelog section as release notes
- Or at minimum, include `gh release create` in the push reminder alongside `git push origin <tag>`

Either way, the skill should surface that GitHub Releases exist as a separate artifact and help the user create one.

### PR DESCRIPTION
**Pull request by cboone**
## Summary

- Add step 11 (Publish) to the release skill that offers to push the commit and tag and create a GitHub Release using the version's changelog section as release notes
- The step is interactive: users can decline and receive manual commands instead
- Handles edge cases for missing remotes, unavailable `gh` CLI, push rejections, and failed release creation
- Bumps release plugin version from 1.1.0 to 1.2.0

## Test plan

- [ ] Run `/release --dry-run` on a test project and verify step 11 is mentioned but not executed
- [ ] Run `/release` on a test project, decline at step 11, verify manual commands are shown (including `gh release create`)
- [ ] Run `/release` on a test project, accept at step 11, verify push and GitHub Release are created
- [ ] Run `check-versions` to verify version consistency across plugin.json and marketplace.json

## Closes

Closes #222

### CODE REVIEW

File: plugins/release/skills/release/SKILL.md [hunk 1]
```diff
@@ -248,5 +323,8 @@ To publish:
 - **CHANGELOG parse error:** If the existing file has an unrecognized format, warn the user and offer to create a new one or append a version section at the top.
 - **Tag already exists:** Abort with a message that the tag `vVERSION` already exists. Suggest choosing a different version.
 - **Not a git repository:** Abort immediately.
-- **No remote configured:** Skip comparison links in CHANGELOG, skip doc version updates, warn the user.
+- **No remote configured:** Skip comparison links in CHANGELOG, skip doc version updates, skip push and GitHub Release in step 11, warn the user.
```
[the remark this answers is not in the document]
Comment (cboone) [reply plugins/release/skills/release/SKILL.md]: Correct. Doc version updates (step 7) are local operations that do not depend on having a remote. Removed "skip doc version updates" from the no-remote error handling. Comparison links and push/release are the only things that depend on a remote. Fixed in 9be2407.

File: plugins/release/skills/release/SKILL.md [hunk 2]
```diff
@@ -221,14 +221,89 @@ Create a GPG-signed annotated tag:
...
+
+#### If the user accepts
+
+##### 11a. Check for remote
+
+```bash
+git remote get-url origin
+```
+
+If no remote is configured, report the error and show the manual push commands without the `gh release create` line. Stop.
```
[the remark this answers is not in the document]
Comment (cboone) [reply plugins/release/skills/release/SKILL.md]: Good catch. Updated the no-remote case to tell the user to configure a Git remote (e.g., `git remote add origin <url>`) before rerunning the step, instead of showing `git push origin ...` commands that would fail. Fixed in 9be2407.
